### ISSUE TITLE
remove rule index from explainability & lint, use rule name

### ISSUE THREAD

**Issue by haim-kermany** [np-guard/vpc-network-config-analyzer#818]

**Comment by ShiriMoran** [comment]
In IBM's VPC:
1. SG rules has ID (but not name); I'll replace index with the ID
2. NACL rules has name; I'll replace index with name

In AWS's VPC
Both SG and NACL do not have neither ID nor Name. This implies that there is no way to identify them but the index; I'll thus keep the index and only add the context of the index (ingress/egress).

@haim-kermany @zivnevo @kyorav

**Comment by kyorav** [comment]
I worry that printing the rule ID might do more harm than good. It looks something like `r006-d10c9e66-7a88-48e6-980e-74cc0ff9d9a4`, making it very hard for humans to parse the results. I think the ID is absolutely needed in order to find the rule only when the SG contains an inordinate number of rules. So maybe we can print just the rule details, and add the rule ID only in a "verbose" output?

**Comment by ShiriMoran** [comment]
This is relevant for explain and for lint
In explain we already have --detail
In lint we already have --printAll
So adding --verbose seems too much

Perhaps we can print for SG the id at the end of the string describing the rule (instead of in the beginning)?

**Comment by ShiriMoran** [comment]
Also, omitting the index for IBM's NACL is problematic since it is the only indication of  prioritization we have.
```
		network ACL acl1-ky allows connection with the following allow and deny rules
			name: acl1-out-1, direction: outbound , src: 10.240.10.0/24 , dst: 10.240.20.0/24, conn: protocol: icmp, action: deny
			name: acl1-out-3, direction: outbound , src: 10.240.10.0/24 , dst: 10.240.20.0/24, conn: all, action: allow
```

I tend to add the index for IBM's NACL and call it "priority"

**Comment by kyorav** [comment]
> Perhaps we can print for SG the id at the end of the string describing the rule (instead of in the beginning)?

This is better, yes.

>So adding --verbose seems too much

This is fine! So we we print the ID only for `--detail` in explain and `--printAll` in lint.
I like this because it means that the regular output will look the same for all cloud providers.

> Also, omitting the index for IBM's NACL is problematic since it is the only indication of prioritization we have

Absolutely. I think for nACLs we should (always) print the index and not print the name, for all providers. The names in IBM are going to be random generated and long anyhow, and the index is meaningful and necessary.

**Comment by kyorav** [comment]
In AWS there is no "Name" field for rules (and almost all other resources). There is a convention of using a tag with key "Name", which the dashboard picks up on and displays. I don't have statistics on how often people use this for rules. If the user does not add a tag, the Name column in the dashboard remains empty -- there will not be a machine-generated name like you see in IBM.

**Comment by zivnevo** [comment]
> aws rules should have names, @zivnevo and I saw them on the dashbord

I now looked again and couldn't find any SG rules with names. We are left with `Security group rule ID`.
SGs however always have their `Security group name` field set, and sometimes also have `Name` tags.

**Comment by ShiriMoran** [comment]
> > Perhaps we can print for SG the id at the end of the string describing the rule (instead of in the beginning)?
> 
> This is better, yes.
> 
> > So adding --verbose seems too much
> 
> This is fine! So we we print the ID only for `--detail` in explain and `--printAll` in lint. I like this because it means that the regular output will look the same for all cloud providers.
> 

In explain the rules are presented only for --detail; so printing the `id` only for detail is the same as printing it always as far as explain is concerned.

> > Also, omitting the index for IBM's NACL is problematic since it is the only indication of prioritization we have
> 
> Absolutely. I think for nACLs we should (always) print the index and not print the name, for all providers. The names in IBM are going to be random generated and long anyhow, and the index is meaningful and necessary.

Having a different format for SG and for NACL is a bit problematic, I'm afraid.
Here's my suggestion: print the name as well as the id in the end of the line; in which case, why not print it always? one can always ignore it

**Comment by ShiriMoran** [comment]
> > aws rules should have names, @zivnevo and I saw them on the dashbord
> 
> I now looked again and couldn't find any SG rules with names. We are left with `Security group rule ID`. SGs however always have their `Security group name` field set, and sometimes also have `Name` tags.

In IBM SG rules do not have names, as implied by the API

**Comment by kyorav** [comment]
>one can always ignore it

I have to disagree on this one. Printing useless information means it is harder to find the relevant information at a glance.

The point of printing the index or the ID is to help the user locate the rule, either in the dashboard or in their terraform. But the IBM dashboard doesn't display the rule ID anywhere, and of course it isn't in the terraform, so it is completely useless information for the user. Rules are uniquely identified by their field values, and that is what users will use to locate the rule. In AWS the IDs are displayed, but I still think it would be easier for me to locate the rule based on the details (ingress, protocol, etc). 

To summarize, I think the ID is useless and should not be printed at all. For nACLs the priority(IBM)/rule no.(AWS) is the best way to identify rules. For SGs people are going to rely on the rule details alone.

**Comment by zivnevo** [comment]
Some dashboard screenshots.
I think we should try to be compliant. One change I would like to have is to have connection details (protocol+ports) as the last column
IBM NACL:
![image](https://github.com/user-attachments/assets/2a67211b-7b41-414a-b733-32bbd81cb117)
IBM SG:
![image](https://github.com/user-attachments/assets/4e34fa30-9d9d-49fd-8f99-df1e7a3ed0ab)
AWS NACL:
![image](https://github.com/user-attachments/assets/acbca36a-0b33-4aaa-b784-7f967c05df13)
AWS SG:
![image](https://github.com/user-attachments/assets/f80471e8-4e65-4462-a112-26000a70d1c9)
